### DISCUSSION TITLE
Yup Schema in Controller Component

### DISCUSSION THREAD

**Discussion by rjhoffmann** [react-hook-form/react-hook-form#7877] [Q&A] [answered]
I'm using the Controller wrapper to render a mui component (TextField, DatePicker, Radio, etc...). I'd like to be able to get the Yup schema for that field, specifically looking to render the component appropriately if it's marked as required in the schema. Is there a way in RHF to do this?

The only thing I can think of currently is to pass the schema definition to the component and inspect it that way.

**Comment by Moshyfawn** [comment, marked as answer]
There's no such field in RHF. And there won't be one, probably, as the resolvers focus on form level validation and not directly associated with the input field. So, your best bet is to pass down your validation schema. At least, the validation field of the input in question.
